ibis/tests/all/test_geospatial.py

```diff
@@ -96,17 +100,91 @@ def test_get_point(backend, geo, expr_fn, expected):
 
 
 @pytest.mark.parametrize(('arg', 'expected'), [(polygon_0, [1.98] * 5)])
-@pytest.mark.only_on_backends([MapD])
+@pytest.mark.only_on_backends(all_db_geo_supported)
 def test_area(backend, geo, arg, expected):
     """Testing for geo spatial area operation."""
     expr = geo[geo, arg.area().name('tmp')]
     result = expr.execute()['tmp']
     testing.assert_almost_equal(result, expected, decimal=2)
 
 
-@pytest.mark.parametrize(('arg', 'expected'), [(point_2.srid() == 4326, True)])
-@pytest.mark.only_on_backends([MapD])
-def test_srid_literals(backend, geo, arg, expected):
-    """Testing for geo spatial srid operation (from literal)."""
-    result = geo[geo, arg.name('tmp')].execute()['tmp'][[0]]
-    testing.assert_almost_equal(result, [expected], decimal=2)
+@pytest.mark.parametrize(
+    ('condition', 'expected'),
+    [
+        (lambda t: point_2.srid(), 4326),
+        (lambda t: point_0.srid(), 0),
+        (lambda t: t.geo_point.srid(), 0),
+        (lambda t: t.geo_linestring.srid(), 0),
+        (lambda t: t.geo_polygon.srid(), 0),
+        (lambda t: t.geo_multipolygon.srid(), 0),
+    ]
+)
+@pytest.mark.only_on_backends(all_db_geo_supported)
+def test_srid(backend, geo, condition, expected):
+    """Testing for geo spatial srid operation."""
+    expr = geo[geo, condition(geo).name('tmp')]
```

Comment:
Do you need the first `geo` selection since you're only pulling out the `tmp` condition for testing?

Reply:
yes, omniscidb doesn't support `select` with just literal.
I can change the `geo` for a specific field if you think it would be better. 

ex: `expr = geo[geo.id, condition(geo).name('tmp')]`